qt/scientific_interfaces/Indirect/test/IndirectFitDataTableModelTest.h

```diff
@@ -0,0 +1,215 @@
+  }
+
+  void test_getnumberOfDomains_returns_total_spectra_of_all_data() {
+    auto dataWorkspace = Mantid::IndirectFitDataCreationHelper::createWorkspace(4, 5);
+    Mantid::API::AnalysisDataService::Instance().addOrReplace("data workspace 2", std::move(dataWorkspace));
+    m_fitData->addWorkspace("data workspace 2");
+    TS_ASSERT_EQUALS(m_fitData->getNumberOfDomains(), 8);
+  }
+
+  void test_getDomainIndex_calculates_correct_value() {
+    auto dataWorkspace = Mantid::IndirectFitDataCreationHelper::createWorkspace(4, 5);
+    Mantid::API::AnalysisDataService::Instance().addOrReplace("data workspace 2", std::move(dataWorkspace));
+    m_fitData->addWorkspace("data workspace 2");
+    TS_ASSERT_EQUALS(m_fitData->getDomainIndex(TableDatasetIndex(0), MantidQt::MantidWidgets::WorkspaceIndex(0)),
+                     FitDomainIndex{0});
+    TS_ASSERT_EQUALS(m_fitData->getDomainIndex(TableDatasetIndex(0), MantidQt::MantidWidgets::WorkspaceIndex(1)),
+                     FitDomainIndex{1});
+    TS_ASSERT_EQUALS(m_fitData->getDomainIndex(TableDatasetIndex(0), MantidQt::MantidWidgets::WorkspaceIndex(2)),
+                     FitDomainIndex{2});
+    TS_ASSERT_EQUALS(m_fitData->getDomainIndex(TableDatasetIndex(0), MantidQt::MantidWidgets::WorkspaceIndex(3)),
+                     FitDomainIndex{3});
+    TS_ASSERT_EQUALS(m_fitData->getDomainIndex(TableDatasetIndex(1), MantidQt::MantidWidgets::WorkspaceIndex(0)),
+                     FitDomainIndex{4});
+  }
+
+  void test_getQValuesForData_returns_correct_value() {
+    auto dataWorkspace = Mantid::IndirectFitDataCreationHelper::createWorkspaceWithInelasticInstrument(4);
+    Mantid::API::AnalysisDataService::Instance().addOrReplace("data workspace 2", dataWorkspace);
+    m_fitData->addWorkspace("data workspace 2");
+    auto spectrumInfo = dataWorkspace->spectrumInfo();
+    auto detID = spectrumInfo.detector(0).getID();
+    double efixed = dataWorkspace->getEFixed(detID);
+    double usignTheta = 0.5 * spectrumInfo.twoTheta(0);
+    double q = Mantid::Kernel::UnitConversion::convertToElasticQ(usignTheta, efixed);
+
+    TS_ASSERT_EQUALS(m_fitData->getQValuesForData()[0], q)
+  }
+
+  void test_that_getResolutionsForFit_return_correctly() {
+    auto resolutionVector = m_fitData->getResolutionsForFit();
+
+    TS_ASSERT_EQUALS(resolutionVector[2].first, "resolution workspace");
+    TS_ASSERT_EQUALS(resolutionVector[2].second, 2);
+  }
+
+  void test_that_getResolutionsForFit_return_correctly_if_resolution_workspace_removed() {
+    Mantid::API::AnalysisDataService::Instance().clear();
+
+    auto resolutionVector = m_fitData->getResolutionsForFit();
+
+    TS_ASSERT_EQUALS(resolutionVector[2].first, "");
+    TS_ASSERT_EQUALS(resolutionVector[2].second, 2);
+  }
+
+  void test_getWorkspaceNames_returns_all_names() {
+    auto dataWorkspace = Mantid::IndirectFitDataCreationHelper::createWorkspace(4, 5);
```

Comment:
`dataWorkspace` seems to be used for the majority of the tests. Would it be worth creating this in the `setUp` method?

Reply:
And also adding it to the ADS in `setUp` too